dev/benchmarks/macrobenchmarks/lib/src/web/recorder.dart

```diff
@@ -377,78 +504,87 @@ abstract class Recorder {
 
     print(
       'SUCCESS: Benchmark converged below ${_kNoiseThreshold * 100}%. '
-      'Noise level is ${profile.drawFrameDurationNoise * 100}%.',
+      'Noise level is $_noiseString.',
     );
     return false;
   }
 
-  Profile _generateProfile() {
-    final List<FrameMetrics> measuredFrames =
-        _frames.sublist(_frames.length - _kMeasuredSampleCount);
-    final Iterable<double> noiseCheckDrawFrameTimes =
-        measuredFrames.map<double>((FrameMetrics metric) =>
-            metric.drawFrameDuration.inMicroseconds.toDouble());
-    final double averageDrawFrameDurationMicros =
-        _computeMean(noiseCheckDrawFrameTimes);
-    final double standardDeviation =
-        _computeStandardDeviationForPopulation(noiseCheckDrawFrameTimes);
-    final double drawFrameDurationNoise =
-        standardDeviation / averageDrawFrameDurationMicros;
-
-    return Profile._(
-      name: name,
-      averageDrawFrameDuration:
-          Duration(microseconds: averageDrawFrameDurationMicros.toInt()),
-      drawFrameDurationNoise: drawFrameDurationNoise,
-      frames: measuredFrames,
-    );
+  @override
+  String toString() {
+    return _formatToStringLines(toDescription());
   }
 }
 
-/// Contains metrics for a series of rendered frames.
+/// Contains metrics for raw benchmark runs that only record durations.
 @immutable
-class Profile {
-  Profile._({
-    @required this.name,
-    @required this.drawFrameDurationNoise,
-    @required this.averageDrawFrameDuration,
-    @required List<FrameMetrics> frames,
-  }) : frames = List<FrameMetrics>.unmodifiable(frames);
+class RawProfile extends Profile<Duration> {
```

Comment:
I'm not sure having a class hierarchy for profiles in addition to the recorder hierarchy is worth the complexity. I think the profile could be one sealed data class. The distinctions between "raw", "frame", "build", etc, could all be encoded in the recorder implementation. This is simpler because you keep related logic (e.g. all "raw" logic) together in one class, and the profile has one common shape.

Having said that, it is clear that the old `Profile` class was too prescriptive. It had too much opinion about frames. I can imagine a few alternative, sufficiently flexible, structures without a class hierarchy:

1. `class Profile { Map<String, List<double>> data; Map<String, double> noises, averages; List<String> scoreKeys }`
1. `class Profile { List<Map<String, double>> data; Map<String, double> noises, averages; List<String> scoreKeys }`
1. `class Profile { Map<String, Timeseries> data; List<String> scoreKeys }`, `class Timeseries { List<double> values, double noise, average }`

Each run of the benchmark records data by appending values to one of the `List<double>`. The `String` in the map labels the metric that we're measuring or computing noise/average from (e.g. `drawFrameDuration`, `windowRenderDuration`, `singleLineTextLayoutDuration`, etc). Each recorder implementation records its own set of labels and values. For example, if you need the list of values recorded for `singleLineTextLayoutDuration` you can get it using `profile.data.values["singleLineTextLayoutDuration"]` (using the 3rd API option).

Reply:
💯 

> For example, if you need the list of values recorded for singleLineTextLayoutDuration you can get it using profile.data.values["singleLineTextLayoutDuration"] (using the 3rd API option).

To make sure I understood this correctly, did you mean `profile.data["singleLineTextLayoutDuration"].values`?

I like the idea of the `Timeseries` class.

Reply:
Yes, `profile.data["singleLineTextLayoutDuration"].values` is what I wanted to say.

Reply:
I think the `Timeseries` option is my favorite too.